### PR TITLE
Changed filename generator

### PR DESCRIPTION
**Pull request by WillTrojak**
This removes the need for a `next` call when the generator is created.

### CODE REVIEW

File: pyfr/util.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def match_paired_paren(delim, n=5):
     return lft*n + mid + rgt*n
 
 
-def file_path_gen(basedir, basename, restore=False):
+def _file_path_gen(basedir, basename, restore=False):
```
Comment (FreddieWitherden) [inline pyfr/util.py:172]: Can probably use an inner function to keep the namespace clean.

**Comment by FreddieWitherden** [comment]
All works okay?

**Comment by WillTrojak** [comment]
Yep, it works fine with the ascent plugin.

**Comment by FreddieWitherden** [comment]
Also, there seem to be conflicts.

**Comment by WillTrojak** [comment]
The conflicts are just from the function being very similar to what was there before and git getting confused.

**Pull request merged**